### DISCUSSION TITLE
vanilo/admin 3.0 requires laravel ^9.0, but it conflicts with root composer require ^11

### DISCUSSION THREAD

**Discussion by DenaByte** [vanilophp/framework#182] [Q&A] [answered]
Hello, I'm doing a fresh install of Vanilo 4.0.
As per the guide, I created a new laravel project with version 11, installed Vanilo 4 with composer then did the other steps from the guide [here](https://vanilo.io/docs/4.x/installation)
Then I proceeded to install vanilo/admin 3.0 and got the following error from composer:

  Problem 1
    - Root composer.json requires vanilo/admin 3.0 -> satisfiable by vanilo/admin[3.0.0].
    - vanilo/admin 3.0.0 requires laravel/framework ^9.0 -> found laravel/framework[v9.0.0, ..., v9.52.16] but it conflicts with your root composer.json require (^11.0).
    
This doesnt make any sense - how does the main framework of vanilo require laravel 10 or 11 at least, but the admin module needs 9 and cant work with 11 or 10 ?
Is there an issue with my composer.json file ?

```json
{
    "name": "laravel/laravel",
    "type": "project",
    "description": "The skeleton application for the Laravel framework.",
    "keywords": ["laravel", "framework"],
    "license": "MIT",
    "require": {
        "php": "^8.2",
        "doctrine/dbal": "^3.5.1",
        "inertiajs/inertia-laravel": "^1.0",
        "laravel/framework": "^11.0",
        "laravel/sanctum": "^4.0",
        "laravel/tinker": "^2.9",
        "tightenco/ziggy": "^2.0",
        "vanilo/framework": "4.0"
    },
    "require-dev": {
        "fakerphp/faker": "^1.23",
        "laravel/breeze": "^2.1",
        "laravel/pint": "^1.13",
        "laravel/sail": "^1.26",
        "mockery/mockery": "^1.6",
        "nunomaduro/collision": "^8.0",
        "phpunit/phpunit": "^10.5",
        "spatie/laravel-ignition": "^2.4"
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ],
        "post-update-cmd": [
            "@php artisan vendor:publish --tag=laravel-assets --ansi --force"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi",
            "@php -r \"file_exists('database/database.sqlite') || touch('database/database.sqlite');\"",
            "@php artisan migrate --ansi"
        ]
    },
    "extra": {
        "branch-alias": {
            "dev-master": "11.x-dev"
        },
        "laravel": {
            "dont-discover": []
        }
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true,
        "allow-plugins": {
            "pestphp/pest-plugin": true,
            "php-http/discovery": true
        }
    },
    "minimum-stability": "stable",
    "prefer-stable": true
}
```

**Comment by fulopattila122** [comment, marked as answer]
The documentation had a typo; it still told you to install Admin v3. It has been fixed.
For Vanilo Framework 4, you need Admin v4, so this is the correct command to install Vanilo Admin:

```bash
composer require vanilo/admin ^4.0
```

**Comment by DenaByte** [comment]
Hello, thanks for the info, i tried with 4.0 and im getting a new error from composer. This is from a new fresh install i just did right now

  Problem 1
    - konekt/appshell 4.0.0 requires cviebrock/eloquent-sluggable ^10.0|^11.0.1 -> found cviebrock/eloquent-sluggable[10.0.0] but the package is fixed to 11.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - vanilo/admin 4.0.0 requires konekt/appshell ^4.0 -> satisfiable by konekt/appshell[4.0.0].
    - Root composer.json requires vanilo/admin 4.0 -> satisfiable by vanilo/admin[4.0.0].

**Reply by fulopattila122** [reply]
The composer message tells exactly what needs to be done:

>  Make sure you list it as an argument for the update command.

ie 

```bash
composer update cviebrock/eloquent-sluggable
composer require vanilo/admin ^4.0
```

**Reply by DenaByte** [reply]
Hello, yes I found another workaround by installing the pack and then manually setting it to vesion 10.0 in composer.json, but looks like yours solution is cleaner.
I dont have a lot of experience with composer, and the versioning is a bit hazy for me, so sorry about that.
I also noticed the documentation is updated to include admin 4.0 so this is fixed and can be closed. Thank you for your time.
